homeassistant/helpers/entity.py

```diff
@@ -253,8 +256,55 @@ def async_update_ha_state(self, force_refresh=False):
                             'https://goo.gl/Nvioub', self.entity_id,
                             end - start)
 
+        def get_overrides(entity_id: str) -> Dict:
```

Comment:
Let's move this method out of the entity. With this much logic, we might want to move all of this to a new customize helper: `homeassistant.helpers.customize`.

In that case we should also start storing the CUSTOMIZE and CACHE dicts in `hass.data`